packages/babel-plugin-transform-block-scoping/test/fixtures/general/switch-callbacks/output.js

```diff
@@ -1,16 +1,17 @@
 function fn() {
+  var _loop = function () {
+    switch (true) {
+      default:
+        var foo = 4;
+        if (true) {
+          var bar = function () {
+            return foo;
+          };
+          console.log(bar());
+        }
+    }
+  };
   while (true) {
-    (function () {
-      switch (true) {
-        default:
-          var foo = 4;
-          if (true) {
-            var bar = function () {
-              return foo;
-            };
-            console.log(bar());
-          }
-      }
-    })();
+    _loop();
```

Comment:
I'm not sure if putting the function declaration inside the loop is better in this case, maybe a little bit of a performance hit, but it improves readability. (especially when there is a complex loop body)
Of course the new behavior is better for me than the old one anyway.

Reply:
Keeping the single `_loop` function also has the advantage over the old behavior that we can avoid multiple closures (example: https://babeljs.io/repl#?browsers=since%202015&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=GYewTgBAFA3DCUEDeAoCEDOB3AlgFwGMALaARkVXXQIEMMBTCUgLjSvQBt68IAPGNuyiIAvAD4-A9AF822fMTIVBEWgyat2nbpJXphEcbpkpZQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=unambiguous&lineWrap=true&presets=env%2Creact%2Cstage-2&prettier=false&targets=&version=7.20.4&externalPlugins=&assumptions=%7B%7D)

And yes, I believe that the function is hoisted for performance reasons, even if it's a small difference "readability of the output" has lower priority :P